ratings and reviews component, on the board.

mattwaelder opened the pull request:
Created boiler plate CSS for group use
installed all fixed layers of reviews / ratings component (no modals)
added api request to base component, storing data to state
passed necessary product info down to respective subcomponents
conditional rendering of some data

although working, i'm not particularly happy with my implementation within react. i ran into issues with api data and storing state, and things just feel messy. down for a code review if anybody wants to give some pointers

yes i do intend to eventually swap to the hooks, but i am scared :)

mattwaelder committed:
added subcomponents folder for each of my modules JSX files
added folder for utilities (api paths, etc)
added basic boiler plate css and some testing css

mattwaelder committed:
added http requests for reviews based on a current product and metadata for the current product
added dynamic rendering of certain data to each review list card
all work currently without hooks, will refactor at later date

mattwaelder committed:
no big changes, just off to dinner

mattwaelder committed:
found a fix for retrieving stars from props in ratings breakdown subcomponent

mattwaelder committed:
stored obj in component state for ratings
rendered review star average and count of votes for each star
added tags to list item
dynamically render method by which list is sorted

mattwaelder committed:
fixed a bug where star counts were rendering in reverse order

mattwaelder commented:
conflicts are just the CSS file, not a biggy at all.

katto030 reviewed (commented):
There are some best practices that I will list here, please consider refactoring to these

React's setState best practice format 
i.e: instead of 
        let stateCopy = this.state;
        stateCopy.reviews = res.data.results;
        this.setState(stateCopy);
write this:
        this.setState({
             reviews: res.data.results
        }) 

 for-loop or .map() instead of hard coding every div
  - you can replace the star number and Object.values(this.props.meta.ratings)[index] if you run a forloop through an array like [5,4,3,2,1]

in ReviewsList.jsx consider directly mapping your components instead of pushing them into an array
  let reviewArr = [];
      this.props.reviews.forEach((el) => {
        reviewArr.push(<ReviewItem review={el} />);
      });
 
you can map directly in <div>{....map((smt) => <Component />}</div>

mattwaelder committed:
implemented mapping for xml render rather than forEach, adjusted calls to set state to be in line with convention

mattwaelder committed:
set my own css file to conform with the group
linked css file within html
fixed bug where avg star count rendered as NaN for the moment before http requests completed

theivanluk reviewed (approved):
Consider making variables more human readable very more efficient code reviews

theivanluk committed:
Merge branch 'main' into main
